packages/twenty-server/src/workspace/utils/deduce-relation-direction.util.ts

```diff
@@ -6,18 +7,32 @@ export enum RelationDirection {
 }
 
 export const deduceRelationDirection = (
-  currentObjectId: string,
+  fieldMetadata: FieldMetadataInterface,
   relationMetadata: RelationMetadataInterface,
 ): RelationDirection => {
-  if (relationMetadata.fromObjectMetadataId === currentObjectId) {
+  if (
+    relationMetadata.fromObjectMetadataId === fieldMetadata.objectMetadataId
+  ) {
+    if (relationMetadata.fromFieldMetadataId === fieldMetadata.id) {
+      return RelationDirection.FROM;
+    }
+
+    if (relationMetadata.toFieldMetadataId === fieldMetadata.id) {
+      return RelationDirection.TO;
+    }
+  }
+
+  if (
+    relationMetadata.fromObjectMetadataId === fieldMetadata.objectMetadataId
+  ) {
     return RelationDirection.FROM;
   }
 
-  if (relationMetadata.toObjectMetadataId === currentObjectId) {
+  if (relationMetadata.toObjectMetadataId === fieldMetadata.objectMetadataId) {
     return RelationDirection.TO;
   }
 
   throw new Error(
-    `Relation metadata ${relationMetadata.id} is not related to object ${currentObjectId}`,
+    `Relation metadata ${relationMetadata.id} is not related to object ${fieldMetadata.objectMetadataId}`,
   );
 };
```

Comment:
Here to proposal for clarity:

Proposal 1:
```typescript
export const deduceRelationDirection = (
  fieldMetadata: FieldMetadataInterface,
  relationMetadata: RelationMetadataInterface,
): RelationDirection => {
  if (relationMetadata.fromObjectMetadataId === fieldMetadata.objectMetadataId) {
    return relationMetadata.fromFieldMetadataId === fieldMetadata.id ? RelationDirection.FROM
         : relationMetadata.toFieldMetadataId === fieldMetadata.id ? RelationDirection.TO
         : RelationDirection.FROM;
  } else if (relationMetadata.toObjectMetadataId === fieldMetadata.objectMetadataId) {
    return RelationDirection.TO;
  }

  throw new Error(
    `Relation metadata ${relationMetadata.id} is not related to object ${fieldMetadata.objectMetadataId}`,
  );
};
```



Proposal 2:
```typescript
export const deduceRelationDirection = (
  fieldMetadata: FieldMetadataInterface,
  relationMetadata: RelationMetadataInterface,
): RelationDirection => {
  if (relationMetadata.fromObjectMetadataId === fieldMetadata.objectMetadataId) {
    if (relationMetadata.fromFieldMetadataId === fieldMetadata.id) {
      return RelationDirection.FROM;
    } else if (relationMetadata.toFieldMetadataId === fieldMetadata.id) {
      return RelationDirection.TO;
    }

    return RelationDirection.FROM;
  }
  
  if (relationMetadata.toObjectMetadataId === fieldMetadata.objectMetadataId) {
    return RelationDirection.TO;
  }

  throw new Error(
    `Relation metadata ${relationMetadata.id} is not related to object ${fieldMetadata.objectMetadataId}`,
  );
};
```

Reply:
I simplified the code and tested:
- object creation
- new field creation
- health check